### ISSUE TITLE
Slow map animation

### ISSUE THREAD

**Issue by bitcraft** [Tuxemon/Tuxemon#361]
Map animations started from the ```play_map_animation``` action are fast on PC, but on mobile/low end platforms (like pocket chip, raspberry pi, etc) contribute to a lot of lag.  Map animations should be cached and handled differently for low end systems.

**Comment by MirceaKitsune** [comment]
The "differently on low end systems" part doesn't sound like a good idea to me; If they are optimized, I believe that optimization should be applied equally on all systems. Unless there are any practical regressions, in which case I support making them an option. Just thought I'd throw this opinion in.

**Comment by bitcraft** [comment]
I didn't mean to say that it should be handled differently when being played on a low end system.  Instead I mean "To accommodate low end systems, the animations should be better optimized".  I'l reword it to make that more clear.

**Comment by Qiangong2** [comment]
Why couldn't we just package it with the animations pre-cached? It might make the package bigger, but it would be speedier

**Comment by bitcraft** [comment]
I think that somewhere in the event engine or where the animations are
loading is causing excessive file system access.  On a computer, with
several GB of RAM, it would be sitting in RAM and never touching the disk
after a few accesses (residing in cache).  On a RAM limited device, it
would likely be a trip to the filesystem each time.  This issue needs some
detailed review, to make sure that the animations are cached properly and
not being loaded excessively.  If the animations are being cached properly,
then some other problem exists...not sure.  The problem is the hang hangs
several ms, enough to make the game lag, each time the grass animation is
played.

Pre-caching of some sort would help, but if it were working correctly, only
the first time the animation is played would cause a slowdown, not
everytime as it is now.

**Comment by bitcraft** [comment]
Ideally, while a map is being loaded, the game will scan the map and load any map animations so they are cached and ready to use.  When loading is complete, remove any unused animations from the cache (which may have been put into cache from old state or map).
